UnhollowerBaseLib/ClassInjector.cs

```diff
@@ -83,11 +85,24 @@ public static bool IsTypeRegisteredInIl2Cpp(Type type)
                     return true;
             return false;
         }
-        
+
         public static void RegisterTypeInIl2Cpp<T>() where T : class => RegisterTypeInIl2Cpp(typeof(T), true, Array.Empty<INativeClassStruct>());
         public static void RegisterTypeInIl2Cpp<T>(bool logSuccess) where T : class => RegisterTypeInIl2Cpp(typeof(T), logSuccess, Array.Empty<INativeClassStruct>());
         public static void RegisterTypeInIl2Cpp(Type type) => RegisterTypeInIl2Cpp(type, true);
         public static void RegisterTypeInIl2Cpp(Type type, bool logSuccess) => RegisterTypeInIl2Cpp(type, logSuccess, Array.Empty<INativeClassStruct>());
+        public static void RegisterTypeInIl2CppWithInterfaces<T>(FindImplementsDelegate fallback = null, bool logSuccess = true) where T : class => RegisterTypeInIl2CppWithInterfaces(typeof(T), fallback, logSuccess);
+        public static void RegisterTypeInIl2CppWithInterfaces(Type type, FindImplementsDelegate fallback, bool logSuccess)
```

Comment:
Imo this is really over complicated, probably would be better to just merge interfaces from `interfaces` argument and interfaces from the attribute in the final overload: https://github.com/BepInEx/Il2CppAssemblyUnhollower/blob/6255d1f1f10a58f49fba92c4ebfbbf21b41d6686/UnhollowerBaseLib/ClassInjector.cs#L119
Generally the whole overload situation was really over complicated even before this PR

Reply:
This is just to have a version with generic and one with Type as parameter as it is currently in ClassInjector.
Imo it is currently overcomplicated because of this I worked with default parameter values instead of thousands of methods with different parameters.

Reply:
This is indeed a bit complicated, and I personally don't like that we now have all these different special overloads

How about adding a separate options object. Something like

```cs
public record RegisterTypeOptions
{
    public string LogSuccess { get; init; } = true;
    public Func<Type, Type[]> InterfacesResolver { get; init; } = null;
}
```

and then have one single overload that contains the main implementation:

```cs
 public static void RegisterTypeInIl2Cpp(Type type, RegisterTypeOptions options) 
```

What do you think about something like this? This allows you to add on new properties to `RegisterTypeOptions` without having to create yet another overload and without breaking binary compatibility as long as `RegisterTypeOptions` has a default value for all properties.

If this sounds good, **for now** it's fine for me if `RegisterTypeOptions` has only those two options and `RegisterTypeInIl2Cpp(Type type, RegisterTypeOptions options)` simply calls into `RegisterTypeInIl2Cpp(Type type, bool logSuccess, params INativeClassStruct[] interfaces)` and then have this new options overload be properly implemented in another PR. Although it would be nice if all values were added at once to the options overload right away so that it would become the new main way to handle injection.

Reply:
This sounds good.
I will change it.